src/app/pages/BlocksPage/index.tsx

```diff
@@ -28,11 +41,18 @@ export const BlocksPage: FC = () => {
     // Listing the latest consensus blocks is not yet implemented.
     // we should call useGetConsensusBlocks()
   }
+
+  useEffect(() => {
+    if (!isMobile) {
+      setTableView(TableLayout.Horizontal)
+    }
+  }, [isMobile, setTableView])
+
   const blocksQuery = useGetRuntimeBlocks<AxiosResponse<TableRuntimeBlockList>>(
     layer, // This is OK, since consensus is already handled separately
     {
-      limit: PAGE_SIZE,
-      offset: offset,
+      limit: tableView === TableLayout.Vertical ? offset || PAGE_SIZE : PAGE_SIZE,
```

Comment:
This doesn't look correct to me.
And it isn't mirrored in transactions

Reply:
Right, fixed.

Reply:
Still doesn't look correct